packages/lib/src/session/signInWithEtheruem.ts

```diff
@@ -0,0 +1,70 @@
+import {
+    createKeyPair,
+    createSigningKeyPair,
+    createStorageKey,
+    getStorageKeyCreationMessage,
+} from '../crypto';
+import { SiweMessage } from 'siwe';
+import { ethers } from 'ethers';
+import { PersonalSign } from '../external-apis/InjectedWeb3API';
+import { ProfileKeys } from '../account';
+
+const DEFAULT_NONCE = 0;
+
+function createSignInWithEthereumMessage(account: string, nonce: number) {
+    const statement = getStorageKeyCreationMessage(nonce);
+    //Create an messgage as defined at EIP-4361
+    const siweMessage = new SiweMessage({
+        //Get domain from widget i guess
+        domain: 'www.dm3.chat',
+        address: account,
+        statement,
+        uri: 'origin',
+        nonce: nonce.toString(),
+        version: '1',
+        chainId: 1,
+        issuedAt: new Date(12).getTime().toString(),
```

Comment:
How should we implement issuedAt ? If we are doing it properly it'll no longer possible to create the encryption key in an deterministic way